toolchain/sem_ir/file.cpp

```diff
@@ -526,15 +526,16 @@ auto GetExprCategory(const File& file, InstId inst_id) -> ExprCategory {
       case FieldDecl::Kind:
       case FunctionDecl::Kind:
       case ImplDecl::Kind:
-      case ImportRefUnloaded::Kind:
       case Namespace::Kind:
       case Return::Kind:
       case ReturnExpr::Kind:
       case StructTypeField::Kind:
         return ExprCategory::NotExpr;
 
-      case CARBON_KIND(ImportRefLoaded inst): {
-        auto import_ir_inst = ir->import_ir_insts().Get(inst.import_ir_inst_id);
+      case ImportRefUnloaded::Kind:
```

Comment:
The handling of this looks correct, but I'm surprised we'd ask for the category of an `ImportRefUnloaded`. Does it make sense that that's happening? It seems a bit scary for code to be looking at the expression category, given that we won't have imported (eg) the type for this case.

Reply:
This causes the error in (reduced example):

```
// --- c1.carbon

library "C1";

class C1 {}

// --- mix_current_package.carbon

library "mix_current_package";

import library "C1";

// --- mix_current_package.impl.carbon

impl library "mix_current_package";

import library "C2";

// CHECK:STDERR: mix_current_package.impl.carbon:[[@LINE+4]]:9: ERROR: Expression cannot be used as a value.
// CHECK:STDERR: var c1: C1 = {};
// CHECK:STDERR:         ^~
// CHECK:STDERR:
var c1: C1 = {};
var c2: C2 = {};
```

I spent a while digging through this, but I've forgotten the actual root cause. Unfortunately I can't investigate quickly (i.e., turn this into a CHECK) because the formatter *also* relies on being able to call GetExprCategory on ImportRefUnloaded.

This cost me the better part of a day to investigate originally, so if you're concerned I'll need to look further next week.

Reply:
I'm not concerned enough to hold up this PR :)

I think we should make sure we understand what's happening, but let's not block on that.